First commits

jsf9k opened the pull request:
## 🗣 Description ##

This pull request contains the first commits for an Ansible role to install the CrowdStrike Falcon sensor.

## 💭 Motivation and context ##

We must install the CrowdStrike Falcon sensor to maintain the COOL ATO.

## 🧪 Testing ##

All automated tests pass.  I was also able to use these changes to build a new OpenVPN AMI for our staging COOL as part of the testing for cisagov/openvpn-packer#87.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
Update Terraform build user code for this Ansible role

jsf9k committed:
Update skeleton for this Ansible role

mcdonnnj commented on README.md:
```diff
@@ -3,16 +3,9 @@
-Actions](https://github.com/features/actions) configurations
-appropriate for an Ansible role, as well as the Terraform code to
-create the AWS test user.
+This is an Ansible role for installing [CrowdStrike
```
This aligns with the language CrowdStrike themselves use in the [installation guide](https://www.crowdstrike.com/blog/tech-center/install-falcon-sensor-for-linux/).

```suggestion
This is an Ansible role for installing the [CrowdStrike
```

mcdonnnj commented on defaults/main.yml:
```diff
@@ -0,0 +1,6 @@
+---
+# The directory where CrowdStrike Falcon sensor is installed
```
```suggestion
# The directory where the CrowdStrike Falcon sensor is installed
```

mcdonnnj commented on README.md:
```diff
@@ -69,18 +62,15 @@ None.
-| optional_variable | Describe its purpose. | `default_value` | No |
-| required_variable | Describe its purpose. | n/a | Yes |
--->
+| install_directory | The directory where CrowdStrike Falcon sensor is installed. | `/opt/CrowdStrike` | No |
```
```suggestion
| install_directory | The directory where the CrowdStrike Falcon sensor is installed. | `/opt/CrowdStrike` | No |
```

mcdonnnj commented on meta/main.yml:
```diff
@@ -1,11 +1,23 @@
+  author: Shane Frasier
   company: CISA Cyber Assessments
-  description: Skeleton Ansible role with test user
+  description: Install CrowdStrike Falcon sensor for CISA CDM
```
```suggestion
  description: Install the CrowdStrike Falcon sensor for CISA CDM
```

mcdonnnj commented on molecule/default/tests/test_default.py:
```diff
@@ -4,15 +4,24 @@
-    """Run a dummy test, just to show what one would look like."""
-    assert x
+def test_falcon_sensor_installed(host):
+    """Test that CrowdStrike Falcon sensor was installed."""
```
```suggestion
    """Test that the CrowdStrike Falcon sensor was installed."""
```

mcdonnnj commented on molecule/default/tests/test_default.py:
```diff
@@ -4,15 +4,24 @@
+
+
+def test_falcon_sensor_enabled(host):
+    """Test that CrowdStrike Falcon sensor is enabled."""
```
```suggestion
    """Test that the CrowdStrike Falcon sensor is enabled."""
```

mcdonnnj commented on tasks/Debian.yml:
```diff
@@ -0,0 +1,4 @@
+---
+- name: Install CrowdStrike Falcon sensor (Debian)
```
```suggestion
- name: Install the CrowdStrike Falcon sensor (Debian)
```

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,77 @@
+        - "{{ role_path }}/vars"
+
+# Check if CrowdStrike Falcon sensor is already installed
+- name: Check if CrowdStrike Falcon sensor is already installed
```
```suggestion
- name: Check if the CrowdStrike Falcon sensor is already installed
```

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,77 @@
+    path: "{{ install_directory }}"
+  register: falcon_directory
+
+- name: Install CrowdStrike Falcon sensor
```
```suggestion
- name: Install the CrowdStrike Falcon sensor
```

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,77 @@
+
+- name: Install CrowdStrike Falcon sensor
+  block:
+    - name: Grab CrowdStrike Falcon sensor system package from S3
```
```suggestion
    - name: Grab the CrowdStrike Falcon sensor system package from S3
```

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,77 @@
+        # Required to set the update_hostname option below
+        # state: started
+
+    # - name: Configure the CrowdStrike Falcon sensor to send hostname updates
```
Comment explaining why this is here but disabled?

jsf9k replied:
Removed in commit 3a01153.

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,77 @@
+    #   failed_when: nessuscli_output.rc != 0
+    #   register: nessuscli_output
+
+    - name: Delete local copy of CrowdStrike Falcon sensor system package
```
```suggestion
    - name: Delete local copy of the CrowdStrike Falcon sensor system package
```

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,77 @@
+      become: no
+      delegate_to: localhost
+
+    - name: Delete remote copy of CrowdStrike Falcon sensor system package
```
```suggestion
    - name: Delete remote copy of the CrowdStrike Falcon sensor system package
```

jsf9k committed:
Refer to CrowdStrike Falcon sensor with a definite article

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove code that is commented out

Ideally we would allow the role user to specify a customer ID for the
CrowdStrike Falcon sensor and simply use a dummy value for testing.
Unfortunately, the customer ID contains a checksum and the falconctl
CLI rejects values with an incorrect checksum; hence I don't know how
to generate a dummy value.

dav3r reviewed (approved):
Looks great! 💪 💼

jsf9k committed:
Remove DHS and CDM certificate dependencies

CrowdStrike Falcon sensor only reaches out to a CrowdStrike-owned web
server, and does not require these certificates or make use of the CDM
IPSec tunnel.

dav3r reviewed (approved):
👍 👍

mcdonnnj commented on meta/main.yml:
Please update the supported platforms to reflect what molecule platforms are still enabled.

jsf9k replied:
Please see commit 7d95f5b.

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,64 @@
+      ansible.builtin.file:
+        path: /tmp/{{ package_object_name }}
+        state: absent
+
+  when: not falcon_directory.stat.exists
```
Something about the extra line bothers me. I'm fine if you decide against it but figured I'd mention it at least.

```suggestion
  when: not falcon_directory.stat.exists
```

mcdonnnj reviewed (changes requested):
One change request and one thought for your consideration.

jsf9k committed:
Remove needless blank line

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj commented on README.md:
```diff
@@ -69,18 +62,15 @@ None.
 ## Dependencies ##
 
-None.
+- [cisagov/ansible-role-cdm-certificates](https://github.com/cisagov/ansible-role-cdm-certificates)
```
There are not in the role metadata. Also does https://github.com/cisagov/openvpn-packer/commit/55c128035e40f7322decd1516825ad1e6704fbba impact this configuration?

mcdonnnj commented on README.md:
```diff
@@ -69,18 +62,15 @@ None.
 ## Dependencies ##
 
-None.
+- [cisagov/ansible-role-cdm-certificates](https://github.com/cisagov/ansible-role-cdm-certificates)
+- [cisagov/ansible-role-dhs-certificates](https://github.com/cisagov/ansible-role-dhs-certificates)
```
There are not in the role metadata. Also does https://github.com/cisagov/openvpn-packer/commit/55c128035e40f7322decd1516825ad1e6704fbba impact this configuration?

jsf9k replied:
Please see commit 3508fdb.

jsf9k committed:
Update role metadata to comment out unsupported platforms

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove role dependencies

I originally thought that the CDM certificates role would be required
for CrowdStrike Falcon to function, but that proved not to be the
case.

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj reviewed (approved):
:shipit:
